main.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"errors"
+	"eventapi/internal/auth"
+	"eventapi/internal/cache"
+	"eventapi/internal/database"
+	"eventapi/internal/middleware"
+	"eventapi/internal/session"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/gofiber/fiber/v2/middleware/compress"
+	"github.com/gofiber/fiber/v2/middleware/encryptcookie"
+)
+
+type ErrorMessage struct {
+	Message string `json:"message"`
+}
+
+func main() {
+	//todo: recover from panics
+	app := fiber.New(fiber.Config{
+		ErrorHandler: func(ctx *fiber.Ctx, err error) error {
+			code := fiber.StatusInternalServerError
+			message := err.Error()
+
+			//todo: unwrap error futher?
+
+			var e *fiber.Error
+
+			if errors.As(err, &e) {
+				code = e.Code
+			}
+
+			return ctx.Status(code).JSON(ErrorMessage{Message: message})
+		},
+	})
+
+	ag := app.Group("api")
+
+	app.Static("/", "./dist")
+
+	ag.Use(compress.New())
+
+	ag.Use(encryptcookie.New(encryptcookie.Config{
+		//todo: find a way to move this to middleware / session service
+		Key:    "session",
+		Except: []string{"csrf_1"},
+	}))
+
+	ag.Use(middleware.UseSession)
+
+	// todo: add csrf header
+	// https://docs.gofiber.io/api/middleware/csrf
+
+	db := database.New()
+	authService := auth.NewAuthService()
+	cacheService := cache.NewCacheService()
+	userRepo := database.NewUserRepo(db)
+	sessionService := session.NewSessionService(cacheService)
```

Comment:
I'd suggest using a configuration library like https://github.com/spf13/viper which supports config files in many different formats, as well as env vars, config structs, and overrides. This way you can define a top level Config struct that has nested structs for the other things you need configuration for (DB, auth service, cache service, etc) and pass those nested structs into each service.